src/runtime/HalideRuntime.h

```diff
@@ -475,21 +474,60 @@ typedef struct buffer_t {
     * replaced with a more general type code in the future. */
     int32_t elem_size;
 
+    /**
+    * Flag bits from halide_buffer_flag_bits_t.
+    */
+    int32_t flags;
+
+    /** An unused field, used to ensure that the struct is an even multiple
+     * of 64 bits on 32-bit systems. */
+    void* _unused;
+} buffer_t;
+
+#endif
+
+#if defined(__cplusplus) && (__cplusplus > 199711L || _MSC_VER >= 1800)
+// Let's assert specific size expectations, because disagreement
+// between caller and callee produces suboptimal results.
+static_assert(sizeof(buffer_t) == (sizeof(void*) == 4 ? 72 : 80),
+              "sizeof(buffer_t) is incorrect");
+#endif
+
+enum halide_buffer_flag_bits_t {
     /** This should be true if there is an existing device allocation
     * mirroring this buffer, and the data has been modified on the
     * host side. */
-    bool host_dirty;
+    halide_buffer_host_dirty = (1 << 0),
 
     /** This should be true if there is an existing device allocation
     mirroring this buffer, and the data has been modified on the
     device side. */
-    bool dev_dirty;
-
-    uint8_t _padding[2];
-} buffer_t;
-#pragma pack(pop)
+    halide_buffer_dev_dirty = (1 << 1)
+};
 
-#endif
+inline bool halide_buffer_get_host_dirty(const buffer_t* buf) {
+    return (buf->flags & halide_buffer_host_dirty) != 0;
+}
+
+inline bool halide_buffer_get_dev_dirty(const buffer_t* buf) {
```

Comment:
Good thought. We'd still need 4 more bytes of padding in 64-bit mode to
ensure the struct is a multiple of 8 bytes.

On Thu, Apr 9, 2015 at 4:34 PM, Zalman Stern <notifications@github.com>
wrote:

> In src/runtime/HalideRuntime.h
> <https://github.com/halide/Halide/pull/737#discussion_r28112548>:
>
> >
> > -#endif
> > +inline bool halide_buffer_get_host_dirty(const buffer_t* buf) {
> > +    return (buf->flags & halide_buffer_host_dirty) != 0;
> > +}
> > +
> > +inline bool halide_buffer_get_dev_dirty(const buffer_t* buf) {
>
> Have we considered just making the flags 32-bit ints?
> -Z- On Apr 9, 2015, at 3:58 PM, Andrew Adams <notifications@github.com>
> wrote: In src/runtime/HalideRuntime.h <
> https://github.com/halide/Halide/pull/737#discussion_r28110775>:
>  -#endif +inline bool halide_buffer_get_host_dirty(const buffer_t* buf) {
> + return (buf->flags & halide_buffer_host_dirty) != 0; +} + +inline bool
> halide_buffer_get_dev_dirty(const buffer_t* buf) {
> Given the issue we had this morning with inline on os x, make sure you
> test apps/glsl — Reply to this email directly or view it on GitHub <
> https://github.com/halide/Halide/pull/737/files#r28110775>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/halide/Halide/pull/737/files#r28112548>.
>